### PR TITLE
Properties: fix and clean

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**

### CODE REVIEW

File: lib/libutee/tee_api_property.c [hunk 1]
```diff
@@ -162,290 +176,274 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
 		res = TEE_ERROR_BAD_PARAMETERS;
-		goto err;
+		goto out;
 	}
 
-	bufferlen = *valueBufferLen;
+	tmp_len = *value_len;
+	if (tmp_len < sizeof(TEE_Identity))
+		tmp_len = sizeof(TEE_Identity);
+	tmp_buf = malloc(tmp_len);
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Aren't we supposed to use `TEE_Malloc()` and friends in this lib?

File: lib/libutee/tee_api_property.c [hunk 2]
```diff
@@ -162,290 +176,274 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
+			*value_len = tmp_len;
+		goto out;
+	}
 
-	switch (pv.type) {
+	switch (type) {
 	case USER_TA_PROP_TYPE_BOOL:
-		l = strlcpy(valueBuffer, pv.u.bool_val ? "true" : "false",
-			    bufferlen);
+		uint32_val = *((unsigned int *)tmp_buf);
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: What if an `unsigned int` is a 64-bit variable?

File: lib/libutee/tee_api_property.c [hunk 3]
```diff
@@ -162,290 +176,274 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
 	case USER_TA_PROP_TYPE_U32:
-		l = snprintf(valueBuffer, bufferlen, "%u",
-			     (unsigned int)pv.u.int_val);
+		uint32_val = *((unsigned int *)tmp_buf);
+		l = snprintf(value, *value_len, "%u", uint32_val);
 		break;
 
 	case USER_TA_PROP_TYPE_UUID:
-		l = snprintf(valueBuffer, bufferlen,
+		uuid_val = *((TEE_UUID *)tmp_buf);
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Why copy the value? Casting the pointer should be enough.

File: lib/libutee/tee_api_property.c [hunk 4]
```diff
@@ -162,290 +176,274 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
-			     pv.u.uuid_val.timeHiAndVersion,
-			     pv.u.uuid_val.clockSeqAndNode[0],
-			     pv.u.uuid_val.clockSeqAndNode[1],
-			     pv.u.uuid_val.clockSeqAndNode[2],
-			     pv.u.uuid_val.clockSeqAndNode[3],
-			     pv.u.uuid_val.clockSeqAndNode[4],
-			     pv.u.uuid_val.clockSeqAndNode[5],
-			     pv.u.uuid_val.clockSeqAndNode[6],
-			     pv.u.uuid_val.clockSeqAndNode[7]);
+			     (unsigned int)uuid_val.timeLow,
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Use proper format code instead of casting.

File: lib/libutee/tee_api_property.c [hunk 5]
```diff
@@ -162,290 +176,274 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
+			     uuid_val.clockSeqAndNode[3],
+			     uuid_val.clockSeqAndNode[4],
+			     uuid_val.clockSeqAndNode[5],
+			     uuid_val.clockSeqAndNode[6],
+			     uuid_val.clockSeqAndNode[7]);
 		break;
 
 	case USER_TA_PROP_TYPE_IDENTITY:
-		l = snprintf(valueBuffer, bufferlen,
+		identity_val = *((TEE_Identity *)tmp_buf);
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Why copy the value? Casting the pointer should be enough.

File: lib/libutee/tee_api_property.c [hunk 6]
```diff
@@ -162,290 +176,274 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
-			     pv.u.identity_val.uuid.timeHiAndVersion,
-			     pv.u.identity_val.uuid.clockSeqAndNode[0],
-			     pv.u.identity_val.uuid.clockSeqAndNode[1],
-			     pv.u.identity_val.uuid.clockSeqAndNode[2],
-			     pv.u.identity_val.uuid.clockSeqAndNode[3],
-			     pv.u.identity_val.uuid.clockSeqAndNode[4],
-			     pv.u.identity_val.uuid.clockSeqAndNode[5],
-			     pv.u.identity_val.uuid.clockSeqAndNode[6],
-			     pv.u.identity_val.uuid.clockSeqAndNode[7]);
+			     (unsigned int)identity_val.login,
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Use proper format code instead of casting.

File: lib/libutee/tee_api_property.c [hunk 7]
```diff
@@ -514,27 +512,26 @@ TEE_Result TEE_GetPropertyName(TEE_PropSetHandle enumerator,
...
+		if (bufferlen >= *name_len)
 			res = TEE_ERROR_SHORT_BUFFER;
-		*nameBufferLen = bufferlen;
+		*name_len = bufferlen;
 	} else {
 		res = utee_get_property((unsigned long)pe->prop_set,
 					pe->idx - eps_len,
-					nameBuffer, nameBufferLen,
-					0, 0, 0);
+					name, name_len, 0, 0, 0);
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: `NULL` is preferable over `0` when assigning pointers.

File: lib/libutee/tee_api_property.c (hunk 4 above)
Comment (jforissier) [inline lib/libutee/tee_api_property.c]: It seems to me that `snprintk("%pUl", (void *)&uuid_val)` may be used here? And same below.

Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: You're right I forgot that it was available for TAs.

Comment (jforissier) [inline lib/libutee/tee_api_property.c]: @jenswi-linaro myself I wasn't sure either and had to check ;-)

File: lib/libutee/tee_api_property.c (hunk 1 above)
Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_property.c]: I think TAs can only use `TEE_Malloc()`, and not `malloc` directly as it is not ensure it exists. However, in libutee, we can use "other" functions, like `malloc`, as we know it is implemented. This is the same for usage of `memset`, which should be `TEE_MemFill` in TAs, or `strlcpy`, which cannot be used in TAs

Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: It's just that mixing `malloc()` with `TEE_Free()` isn't working as `TEE_Malloc()` adds a small header to store the inconvenient `hint`.

**Comment by pascal-brand-st-dev** [comment]
Updated

File: lib/libutee/tee_api_property.c [hunk 8]
```diff
@@ -162,290 +177,250 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
+		goto out;
+	}
 
-	switch (pv.type) {
+	switch (type) {
 	case USER_TA_PROP_TYPE_BOOL:
-		l = strlcpy(valueBuffer, pv.u.bool_val ? "true" : "false",
-			    bufferlen);
+		uint32_val = *((uint32_t *)tmp_buf);
+		l = strlcpy(value, (uint32_val ? "true" : "false"),
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: `l--` is needed after this assignment.

File: lib/libutee/tee_api_property.c [hunk 9]
```diff
@@ -162,290 +177,250 @@ static TEE_Result propget_get_property(TEE_PropSetHandle h, char *name,
...
-			     pv.u.identity_val.uuid.clockSeqAndNode[6],
-			     pv.u.identity_val.uuid.clockSeqAndNode[7]);
+		p_identity_val = ((TEE_Identity *)tmp_buf);
+		l = snprintk(value, *value_len, "%u:%pUl",
+			     p_identity_val->login,
+			     (void *)(&(p_identity_val->uuid)));
 		break;
 
 	case USER_TA_PROP_TYPE_STRING:
+		l = strlcpy(value, tmp_buf, *value_len);
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: `l--` is needed after this assignment.

File: lib/libutee/tee_api_property.c (hunk 8 above)
Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_property.c]: `strlcpy` does not count the last 0 zero termination. So I do not think we need `l--` after the assignment.

Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: OK, my bad.

File: lib/libutee/tee_api_property.c [hunk 10]
```diff
@@ -514,27 +489,26 @@ TEE_Result TEE_GetPropertyName(TEE_PropSetHandle enumerator,
...
 
 	if (pe->idx < eps_len) {
 		str = eps[pe->idx].name;
-		bufferlen = strlcpy(nameBuffer, str, *nameBufferLen);
-		if (bufferlen >= *nameBufferLen)
+		bufferlen = strlcpy(name, str, *name_len);
+		if (bufferlen >= *name_len)
 			res = TEE_ERROR_SHORT_BUFFER;
-		*nameBufferLen = bufferlen;
+		*name_len = bufferlen;
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Since `strlcpy()` doesn't count the zero termination we need to do `+ 1` here.

**Comment by pascal-brand-st-dev** [comment]
Fixed

File: lib/libutee/tee_api_property.c [hunk 11]
```diff
@@ -514,27 +489,26 @@ TEE_Result TEE_GetPropertyName(TEE_PropSetHandle enumerator,
...
 	res = propset_get(pe->prop_set, &eps, &eps_len);
 	if (res != TEE_SUCCESS)
 		goto err;
 
 	if (pe->idx < eps_len) {
 		str = eps[pe->idx].name;
-		bufferlen = strlcpy(nameBuffer, str, *nameBufferLen);
-		if (bufferlen >= *nameBufferLen)
+		bufferlen = strlcpy(name, str, *name_len) + 1;
+		if (bufferlen >= *name_len)
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api_property.c]: Should be '>' instead of '>=' since we've already added `1`

Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_property.c]: :angry: sory for that. I'll modify the test too to get this case

**Comment by pascal-brand-st-dev** [comment]
updated

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**